openapi-cli/src/test/java/io/ballerina/openapi/generators/openapi/NegativeConstraintTests.java

```diff
@@ -49,4 +55,64 @@ public void testInvalidRegexPatterns() throws IOException {
+        for (int i = 0; i < errors.size(); i++) {
+            Assert.assertEquals(errors.get(i).getMessage(), "Generated OpenAPI definition does not contain" +
+                    " variable assignment '" + expectedVariables.get(i) + "' in constraint validation.");
+        }
+    }
+
+    /*
+     * This test is used to check whether it gives warnings when '@constraint:Date' is being used.
+     * Currently, we don't have support for mapping Date constraints to OAS hence we skip them.
+     * TODO: <a href="https://github.com/ballerina-platform/ballerina-standard-library/issues/5049">...</a>
+     * Once the above improvement is completed this Negative test should be removed and should add as a Unit test!
+     */
+    @Test(description = "when the record field has time:Date record type")
+    public void testDateType() throws IOException {
+        Path ballerinaFilePath = RES_DIR.resolve("constraint-negative/negative_date.bal");
+        List<OpenAPIConverterDiagnostic> errors = TestUtils.compareWithGeneratedFile(new OASContractGenerator(),
+                ballerinaFilePath, "constraint-negative/negative_date.yaml");
+        errors.forEach(error -> Assert.assertEquals(error.getMessage(), "Ballerina Date constraints might " +
+                "not be reflected in the OpenAPI definition"));
+    }
+
+    /*
+     * This test is used to check whether it gives compilation errors when invalid REGEX patterns are given.
+     * To get more context go through the referenced links.
+     * By fixing below Bug will make this test fail.
+     * {@link <a href="https://github.com/ballerina-platform/ballerina-lang/issues/41492">...</a>}
+     * TODO: <a href="https://github.com/ballerina-platform/ballerina-standard-library/issues/5048">...</a>
+     */
+    @Test(description = "When String constraint has compilation errors (REGEX pattern with invalid format)")
+    public void testInvalidRegexPattern() {
+        Path ballerinaFilePath = RES_DIR.resolve("constraint-negative/invalidRegexPattern.bal");
+        Project project = getProject(ballerinaFilePath);
+        DiagnosticResult diagnostic = getCompilation(project);
+        Object[] errors = diagnostic.diagnostics().stream().filter(d ->
+                DiagnosticSeverity.ERROR == d.diagnosticInfo().severity()).toArray();
+        Assert.assertEquals(errors.length, 2);
+        Assert.assertTrue(errors[0].toString().contains("ERROR [invalidRegexPattern.bal:(21:20,21:20)] " +
+                "invalid char after backslash"));
+        Assert.assertTrue(errors[1].toString().contains("ERROR [invalidRegexPattern.bal:(21:22,21:22)] " +
+                "missing backslash"));
+    }
+
+    @Test(description = "When Integer constraint has float value which is invalid")
+    public void testInvalidInteger() {
+        Path ballerinaFilePath = RES_DIR.resolve("constraint-negative/invalidInteger.bal");
+        Project project = getProject(ballerinaFilePath);
```

Comment:
```suggestion
    public void testInvalidInteger() throws ProjectException {
        Path ballerinaFilePath = RES_DIR.resolve("constraint-negative/invalidInteger.bal");
        Project project = ProjectLoader.loadProject(servicePath);
```

Check for other places as well.

Reply:
Resolved